lib/visitor/strategy/default_strategy.dart

```diff
@@ -53,10 +53,19 @@ class DefaultStrategy implements IVisitor {
     }
   }
 
+  @override
+  // Get Modification object, this object will be used by the flag class
+  Modification? getFlagModification(String key) {
+    return visitor.modifications[key];
+  }
+
   @override
   T getModification<T>(String key, T defaultValue, {bool activate = false}) {
     var ret = defaultValue;
 
+    bool hasSameType =
```

Comment:
It would be simpler I think, if `hasSameType` was true by default, but false in the default case of the switch. 
It would prevent all the `hasSameType = true;` redundant lines